### PR TITLE
Added NavLink for functional routing

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
## Summary

Users need to be able to navigate between the pages of our application in order to access all of its features. There is a navigation bar in the `Layout.jsx` component, but it's not quite feature-complete: the links in the nav element don't actually allow the user to navigate from page to page.

## Acceptance criteria
- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Notes
- The [official docs for the `NavLink` component](https://reactrouterdotcom.fly.dev/docs/en/v6/components/nav-link) will come in very handy!
- While a standard anchor element (`<a>`) _could_ allow a user to navigate from page to page, the `NavLink` component adds additional behavior that is essential to our app functioning correctly!

### PR DESCRIPTION
**Pull request by Yaosaur**
## Description

This PR adds functional routing via NavLink components for the pages we have and shows which link is currently active.

## Related Issue

- https://github.com/the-collab-lab/tcl-55-smart-shopping-list/issues/2

## Acceptance Criteria

- [x] The nav element is added to the Layout component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the nav element
- [x] The links all function as expected using the NavLink component from react-router-dom

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |

### After

image

## Testing Steps / QA Criteria

Click on navbar links on the bottom of the page. The link you just clicked on should be red to reflect the now active page.

### CODE REVIEW

**Comment by lyleschemmerling** [comment]
Good catch on the styling change

**Review by katherineyuneman — APPROVED** [review]
We were curious about removal of the styling piece but Yao clarified at Office Hours.  LGTM - everything functioning as it is supposed to based on AC and code is clean and readable!

**Review by jeremiahfallin — APPROVED** [review]
Routing works as expected, lgtm.

**Comment by mentalcaries** [comment]
Nice work!!

**Comment by jeremiahfallin** [comment]
Just checked and if you change the NavLink lines as so:
```<NavLink to="/" className="Nav-link">Home</NavLink>
<NavLink to="/list" className="Nav-link">List</NavLink>
<NavLink to="/add-item" className="Nav-link">Add Item</NavLink>
```

and revert the changes to the `css` file then the active page link shows as intended. This way we don't have to stop using our `css variables`.

Also it might make since to group our imports from the same library:
`import { NavLink, Outlet } from 'react-router-dom';`

**Comment by mentalcaries** [comment]
On testing, I had the same experience as @jeremiahfallin with respect to the active styling.

Whether you choose to use the original styles is up to you at this point (knowing that we will address the appearance of the app later on). Worth considering would be to add a `color` property to the new `Nav` style so it doesn't have the 'visited' appearance assigned. Again, this is strictly optional as styles are out of the project scope right now.

I do think you should consolidate the imports from `react-router-dom` though.

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Outlet } from 'react-router-dom';
+import { NavLink } from 'react-router-dom';
```
Comment (mentalcaries) [inline src/views/Layout.jsx:2]: You can combine these into a single import

Comment (Yaosaur) [reply src/views/Layout.jsx:2]: Thanks for pointing that out!

**Comment by Yaosaur** [comment]
> On testing, I had the same experience as @jeremiahfallin with respect to the active styling.
> 
> Whether you choose to use the original styles is up to you at this point (knowing that we will address the appearance of the app later on). Worth considering would be to add a `color` property to the new `Nav` style so it doesn't have the 'visited' appearance assigned. Again, this is strictly optional as styles are out of the project scope right now.
> 
> I do think you should consolidate the imports from `react-router-dom` though.

Thank you Jeremiah and Devin for pointing out the styling. I didn't think about using className Nav-link on the component with the same link, but it makes sense to do so since changing styles is out of scope now. With the current styles the visited links don't have the 'visited' appearance.

**Review by lyleschemmerling — APPROVED** [review]

**Review by 5hraddha — APPROVED** [review]

**Pull request merged**